File: leetcode/String/leetcode-58/58.java
class Solution {
    public int lengthOfLastWord(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        int res = 0;

        int i = s.length() - 1;
        while (i >= 0 && s.charAt(i) == ' ') {
            i--;
        }

        while (i >= 0 && s.charAt(i) != ' ') {
            res++;
            i--;
        }
        return res;
    }
}

File: leetcode/String/leetcode-58/58.go
package string

func lengthOfLastWord(s string) int {
	if len(s) == 0 {
		return 0
	}

	res := 0
	i := len(s) - 1

	for i >= 0 && s[i] == ' ' {
		i--
	}

	for i >= 0 && s[i] != ' ' {
		i--
		res++
	}
	return res
}
